src/Microsoft.OData.Client/Serialization/HttpWebRequestMessage.cs

```diff
@@ -239,26 +306,19 @@ public override void SetHeader(string headerName, string headerValue)
         /// <returns>Stream to which the request payload needs to be written.</returns>
         public override Stream GetStream()
         {
-#if PORTABLELIB
-            // not supported in async environments. Another IODataRequestMessage which buffers the request should be used.
-            throw new NotSupportedException(Strings.ODataRequestMessage_GetStreamMethodNotSupported);
-#else
             if (this.inSendingRequest2Event)
             {
                 throw new NotSupportedException(Strings.ODataRequestMessage_GetStreamMethodNotSupported);
             }
-
-            return this.httpRequest.GetRequestStream();
-#endif
+            return _messageStream;

```

Comment:
Just a question is there a chance that calling this early may result in the stream being empty as the request has not been sent yet? Or is this the stream that is being used to write the request payload?

Reply:
yes this is the stream used to write the request payload.